darts/models/forecasting_model.py

```diff
@@ -28,54 +32,83 @@ class ForecastingModel(ABC):
     """ The base class for all forecasting models.
 
     All implementations of forecasting have to implement the `fit()` and `predict()` methods defined below.
+    The `fit()` method is meant to train the model on one training time series.
+
+    In addition, models can optionally implement the `multi_fit()` method. This method is meant to train the model
+    on several time series, or to handle cases where the "features" and "target" dimensions are not the same.
+    `fit()` will typically be the entry point for quickly fitting reasonably simple models to one series,
+     while `multi_fit()` will typically be the path taken by the more sophisticated machine learning models.
```

Comment:
Combining support for multiple time series input with the separation of input vs target series since both are unique to torch models makes a lot of sense to me!